src/Orleans.Core.Abstractions/Exceptions/OrleansException.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// An exception class used by the Orleans runtime for reporting errors.
+    /// </summary>
+    /// <remarks>
+    /// This is also the base class for any more specific exceptions 
+    /// raised by the Orleans runtime.
+    /// </remarks>
+    [Serializable]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1058:TypesShouldNotExtendCertainBaseTypes")]
```

Comment:
I think this is not needed, as `Exception` is fine to derive from. Can be dealt with in a separate PR though